### PR TITLE
Pull group plugin movement fix

### PR DESCRIPTION
**Pull request by nickkrstevski**
Keeps groups static when using pull group kicad plugin
- Calculates anchor component in group based on pin count (component area tie breaker)
- Calculates source vector from origin
- Calculates target vector from origin
- Applied target_vector - source_vector translation to entire group after all layout info is updated to source positions

### CODE REVIEW

**Comment by mawildoer** [comment]
Hell yeah brother! 👏

I’ll get to reviewing this properly come morning, but I love it already!!

**Comment by nickkrstevski** [comment]
Woohoo!

Fix #215

**Comment by napowderly** [comment]
Tested! Working for me on Kicad8

File: src/atopile/kicad_plugin/common.py [hunk 1]
```diff
@@ -91,3 +91,41 @@ def sync_footprints(
...
+    fps = [item for item in items if isinstance(item, pcbnew.FOOTPRINT)]
+    for fp in fps:
+        if fp.GetPadCount() > max_padcount or (fp.GetPadCount()==max_padcount and fp.GetArea()>max_area):
+            anchor_fp = fp
+            max_padcount = fp.GetPadCount()
+            max_area = fp.GetArea()
+    return anchor_fp
+
+
+def find_anchor_footprint_board(board: pcbnew.BOARD) -> pcbnew.FOOTPRINT:
```
Comment (mawildoer) [inline src/atopile/kicad_plugin/common.py:111]: We could potentially consolidate these functions? Something for the future!

Comment (nickkrstevski) [reply src/atopile/kicad_plugin/common.py:111]: Nice, I just did this in the pushgroup fix PR

**Pull request merged**